hail_v02_scripts/utils/elasticsearch/client.py

```diff
@@ -0,0 +1,253 @@
+            elasticsearch_config["es.write.rest.error.handlers"] = "log"
+            elasticsearch_config["es.write.rest.error.handler.log.logger.name"] = "BulkErrors"
+
+        # encode any special chars in column names
+        rename_dict = {}
+        for field_name in table.row_value.dtype.fields:
+            encoded_name = field_name
+
+            # optionally replace . with _ in a non-reversible way
+            if field_names_replace_dot_with is not None:
+                encoded_name = encoded_name.replace(".", field_names_replace_dot_with)
+
+            # replace all other special chars with an encoding that's uglier, but reversible
+            encoded_name = encode_field_name(encoded_name)
+
+            if encoded_name != field_name:
+                rename_dict[field_name] = encoded_name
+
+        for original_name, encoded_name in rename_dict.items():
+            logger.info("Encoding column name %s to %s" % (original_name, encoded_name))
+
+        table = table.rename(rename_dict)
+
+        if verbose:
+            logger.info(pformat(table.row_value.dtype))
+
+        # create elasticsearch index with fields that match the ones in the table
+        elasticsearch_schema = elasticsearch_schema_for_table(
+            table,
+            disable_doc_values_for_fields=disable_doc_values_for_fields,
+            disable_index_for_fields=disable_index_for_fields,
+        )
+
+        # override elasticsearch types
+        if field_name_to_elasticsearch_type_map is not None:
+            modified_elasticsearch_schema = dict(elasticsearch_schema)  # make a copy
+            for field_name_regexp, elasticsearch_field_spec in field_name_to_elasticsearch_type_map.items():
+                match_count = 0
+                for key in elasticsearch_schema.keys():
+                    if re.match(field_name_regexp, key):
+                        modified_elasticsearch_schema[key] = elasticsearch_field_spec
+                        match_count += 1
+
+                logger.info("%s columns matched '%s'" % (match_count, field_name_regexp))
+
+            elasticsearch_schema = modified_elasticsearch_schema
+
+        # optionally delete the index before creating it
+        if delete_index_before_exporting and self.es.indices.exists(index=index_name):
+            self.es.indices.delete(index=index_name)
+
+        _meta = None
+        if export_globals_to_index_meta:
+            _meta = dict(table.globals.value)
+
+        self.create_or_update_mapping(
+            index_name, index_type_name, elasticsearch_schema, num_shards=num_shards, _meta=_meta
+        )
+
+        logger.info(
+            "==> exporting data to elasticsearch. Write mode: %s, blocksize: %s"
+            % (elasticsearch_write_operation, block_size)
+        )
+
+        # FIXME: https://github.com/hail-is/hail/issues/4063
+        elasticsearch_config = None
```

Comment:
For now, the following arguments won't work because they use this configuration.
* elasticsearch_write_operation
* ignore_elasticsearch_write_errors
* elasticsearch_mapping_id
* disable_doc_values_for_fields
* disable_index_for_fields

Reply:
This Hail bug has been fixed, so this can be removed now.